### ISSUE TITLE
Destructuring `let` attribute in `<For>` component

### ISSUE THREAD

**Issue by bicarlsen** [leptos-rs/leptos#3059]
**Is your feature request related to a problem? Please describe.**
When iterating over a structured type (i.e. a `tuple` or `struct`) in a `<For>` component, it would be convenient if we could destructure it in the `let` attribute. To do this at the moment requires additional boilerplate.
e.g. 
```rust
struct Item {
  pub name: String,
  pub value: String,
}

#[component]
fn List(list: Vec<Item>) -> impl IntoView {
  view! {
    <For
      each=move || list
      key=|(name, _)| name.clone()
      let:item
    >
      {move || { // Boilerplate to destructure
        let Item { name, value } = &item;
        view! {
          <div>{name} ": " {description}</div>
        }
      }}
    </For>
  }
}
```
```rust
type Point = (i32, i32);

#[component]
fn Points(points: Vec<Point>) -> impl IntoView {
  view! {
    <For
      each=move || points
      key=|point| point
      let:point
    >
      {move || { // Boilerplate to destructure
        let (x, y) = point;
        view! {
          <div>{x} ", " {y}</div>
        }
      }}
    </For>
  }
}
```

**Describe the solution you'd like**
It would be nice to allow destructuring in the `let` attribute to remove this additional layer of code.
```rust
#[component]
fn List(list: Vec<Item>) -> impl IntoView {
  view! {
    <For
      each=move || list
      key=|(name, _)| name.clone()
      let:Item { name, value }  // Destructure in `let`
    >
      <div>{name} ": " {description}</div>
    </For>
  }
}
```
```rust
#[component]
fn Points(points: Vec<Point>) -> impl IntoView {
  view! {
    <For
      each=move || points
      key=|point| point
      let:(x, y)  // Destructure in `let`
    >
      <div>{x} ", " {y}</div>
    </For>
  }
}
```

**Comment by gbj** [comment]
There's a syntax for this implemented by #2655. The below compiles. Note the `let(/* pattern */)` syntax.
```rust
#[derive(Clone)]
struct Item {
    pub name: String,
    pub value: String,
}

#[component]
fn List(list: ReadSignal<Vec<Item>>) -> impl IntoView {
    view! {
      <For
        each=move || list.get()
        key=|Item { name, .. }| name.clone()
        let(Item { name, value })
      >
        <div>{name} ": " {value}</div>
      </For>
    }
}
```

**Comment by bicarlsen** [comment]
Awesome!
